src/components/Popup/OfficePopup.vue

```diff
@@ -0,0 +1,96 @@
+<template>
+    <div
+            ref="self"
+            :class="classNames.root"
+            @focus="focused = true"
+            @blur="onSelfBlur"
+    >
+        <slot/>
+    </div>
+</template>
+
+<script lang="ts">
+    import {Component, Vue, Prop} from "vue-property-decorator";
+    import {mergeStyleSets} from "@uifabric/merge-styles";
+    import {getDocument} from "@/utility/dom";
+
+    @Component
+    export default class OfficePopup extends Vue {
+        @Prop({type: Boolean, default: true}) private needsVerticalScrollBar!: boolean;
+        @Prop({type: Boolean, default: false}) private shouldRestoreFocus!: boolean;
+        private originalFocusedElement?: HTMLElement;
+
+        private focused: boolean = false;
+
+        get classNames() {
+            return mergeStyleSets({
+                root: [
+                    this.needsVerticalScrollBar && {
+                        overflowY: "scroll"
+                    },
+                    !this.needsVerticalScrollBar && {
+                        overflowY: undefined
+                    }]
+            });
+        }
+
+        public beforeMount() {
+            this.originalFocusedElement = getDocument()!.activeElement as HTMLElement;
+        }
+
+        public mounted() {
+            this.updateScrollBar();
+        }
+
+        public updated() {
+            this.updateScrollBar();
+        }
+
+        public beforeDestroy() {
+            if (this.shouldRestoreFocus &&
+                this.originalFocusedElement &&
+                this.focused &&
+                (this.originalFocusedElement as any) !== window &&
+                this.originalFocusedElement) {
+                this.originalFocusedElement.focus();
+            }
+        }
+
+        public onSelfBlur(event: any) {
+            if ((this.$refs.self as HTMLElement).contains(event.relatedTarget as HTMLElement)) {
+                this.focused = false;
+            }
+        }
+
+        private updateScrollBar() {
+            requestAnimationFrame(() => this.getScrollBar());
```

Comment:
try it without the `requestAnimationFrame` and see if something changes. I'm not sure if we need this in a Vue.js environment.

Reply:
requestAnimationFrame is needed to perform the operation before / for the next repaint.

Reply:
Yes I got this, but do we really need this in Vue.js? The Vue Framework should handle this kind of stuff for us.